### DISCUSSION TITLE
Will we plan to build a corresponding TS/JS library to enable using the composable architecture in frontend dev?

### DISCUSSION THREAD

**Discussion by RutaTang** [pointfreeco/swift-composable-architecture#2478] [Q&A] [answered]
TCA is really amazing and smooth and I think I can not go back to the other architectures. Thus, I plan to use TCA in my frontend project but I still can not find a library that can enable me to use TCA, even Redux which seems not enabling me to use `scope` to scope the store.

Thus, really hope there is a kind of library that implements TCA concepts for frontend dev.

**Comment by kamcma** [comment]
I've been out of the web front-end game for a couple years so I don't know the latest status of things, but the name TCA is an homage to T _E_ A, [the Elm architecture](https://guide.elm-lang.org/architecture/), a web front-end language-and-architecture, and one of the big early popularizers of functional, unidirectional architectures. You could look at that.

I'm sure there are others.

**Reply by stephencelis** [reply, marked as answer]
@kamcma Sadly I don't think Elm or other Redux-style architectures focus on testing and composition of features to the same extent as TCA. There are some very basic means of combining reducers and general function composition folks rely on, but we don't know of any libraries out there that offers the story we've come to.

That said we would love for TCA to work on other platforms, including the web via Wasm, and we hope to get there some day, but it will take some time.

**Reply by RutaTang** [reply]
I have read the Elm guide but I hope I can use TCA or TEA in TS/JS instead of using Elm so that I can easily integrate other other TS/JS lib. As stephencelis said, I have worked with Redux for several projects but I can not find a way to imitate TCA with it and I even researched outed other popular state management lib, but they seems also lack ability mentioned in TCA to do composition, better unit testing and inject dependencies.

Really hope TCA can come to the Web and thanks for your great work!

**Reply by saroar** [reply]
TCA with web that will be game change :) just boom 🚀
